stats/src/main/java/com/facebook/stats/mx/StatsReader.java

```diff
@@ -36,6 +36,8 @@
   public MultiWindowDistribution getDistribution(String key);
   public String getAttribute(StatType key);
   public String getAttribute(String key);
+  public Callable<Long> getDynamicCounter(StatType key);
+  public Callable<Long> getDynamicCounter(String key);
```

Comment:
i didn’t realize there wasn’t any way at all to get a dynamic coutner directly (diff name makes sense now)

i think returning the counter value is more consistent with the attribute API

also, i sort of think that making get/add counter work like they are in the same hash (ie same namespace)

not sure how hard this is offhand: we’d need to unify the two counter hashes or do a check if they exist in both first (requires locking, bleah)

probably this way (either with the Callable or just the value) is best


From: Matt Dordal <notifications@github.com<mailto:notifications@github.com>>
Reply-To: facebook/jcommon <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, March 10, 2015 at 1:10 PM
To: facebook/jcommon <jcommon@noreply.github.com<mailto:jcommon@noreply.github.com>>
Cc: Internal Use <rash@fb.com<mailto:rash@fb.com>>
Subject: Re: [jcommon] Add way to get dynamic counters directly from the stats object (#32)


In stats/src/main/java/com/facebook/stats/mx/StatsReader.java<https://github.com/facebook/jcommon/pull/32#discussion_r26166539>:

> @@ -36,6 +36,8 @@
>    public MultiWindowDistribution getDistribution(String key);
>    public String getAttribute(StatType key);
>    public String getAttribute(String key);
> +  public Callable<Long> getDynamicCounter(StatType key);
> +  public Callable<Long> getDynamicCounter(String key);


I don't have any need for the Callable<> directly, but the other getFooCounter things returned the underlying object. Would you prefer it just return the Long?

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/jcommon/pull/32/files#r26166539>.